drivers/lora/sx126x.c

```diff
@@ -56,10 +56,17 @@ BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(semtech_sx1261) +
 #define GPIO_ANTENNA_ENABLE_FLAGS			\
 	DT_INST_GPIO_FLAGS(0, antenna_enable_gpios)
 
-#define DIO2_TX_ENABLE DT_INST_PROP(0, dio2_tx_enable)
+#define HAVE_GPIO_TX_ENABLE	DT_INST_NODE_HAS_PROP(0, tx_enable_gpios)
```

Comment:
I wonder how much we get by using `HAVE_GPIO_TX_ENABLE` instead of `DT_INST_NODE_HAS_PROP(0, tx_enable_gpios)` directly. Not much in terms of characters used. The main advantage is maybe missing symbol in case of typos.

When reviewing your code I ran into another idea, presented here: #26618. The same might be considered for sx1262 (if reviewers will like that solution). At least flags and label macros do not have to be redefined.

Reply:
Agreed, it doesn't same much in terms of typing. My main rationale was that it keeps the intention of the code clearer by keeping the DT bindings, including validation, in one place.

Reply:
Your proposed solution looks really nice. Let's save that for another PR though. :)

Reply:
Okay ;)